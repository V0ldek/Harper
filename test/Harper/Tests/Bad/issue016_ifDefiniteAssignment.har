f :: Integer -> Integer;
f n = {
  var (x :: Integer);
  
  if (n == 42) {
    x := n;
  }

  (result :: Integer) = x + n;
  return result;
};

main :: Integer;
main = f 42;