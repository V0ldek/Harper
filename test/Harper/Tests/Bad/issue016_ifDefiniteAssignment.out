Error: variable `x` is not definitely assigned at point of use.
During evaluation of:
  x
Located at line 9 column 25

Execution terminated with an error: type error.